DEVELOPER_GUIDE.md

```diff
@@ -0,0 +1,66 @@
+# Developer Guide
+
+So you want to contribute code to OpenSearch Go Client?  Excellent!  We're glad you're here.  Here's what you need to do:
```

Comment:
nit: Should it be "So you want to contribute code to the OpenSearch Go Client?"

Reply:
Ack